features/netsocket/emac-drivers/TARGET_ARM_SSG/COMPONENT_SMSC9220/smsc9220_emac.cpp

```diff
@@ -87,12 +88,19 @@ emac_mem_buf_t *SMSC9220_EMAC::low_level_input()
         message_length -= CRC_LENGTH_BYTES;
     }
 
-    p = _memory_manager->alloc_heap(message_length, SMSC9220_BUFF_ALIGNMENT);
+    p = _memory_manager->alloc_heap(SMSC9220_ETH_MAX_FRAME_SIZE,
+                                    SMSC9220_BUFF_ALIGNMENT);
 
     if (p != NULL) {
         _RXLockMutex.lock();
-        smsc9220_receive_by_chunks(dev, (char*)_memory_manager->get_ptr(p),
-                                   _memory_manager->get_len(p));
+        received_bytes = smsc9220_receive_by_chunks(dev,
+                                  (char*)_memory_manager->get_ptr(p),
+                                  _memory_manager->get_len(p));
+        if(received_bytes == 0){
+            _memory_manager->free(p);
```

Comment:
This leaves `p` invalid. Presumably should be followed by `p = nullptr;`.

Reply:
Added p = nullptr; as suggested.